packages/headless/src/controllers/result-list/headless-result-list.ts

```diff
@@ -51,6 +51,31 @@ export function buildResultList(
     dispatch(registerFieldsToInclude(options.fieldsToInclude));
   }
 
+  let lastFetchCompleted = 0;
+  let consecutiveFetches = 0;
+  const maxConsecutiveFetches = 5;
+  const maxDelayBetweenFetches = 200;
```

Comment:
Would a better name for this variable be `minDelay...`? I think this reads better: if the current delay is less than the minimum delay, then increment the counter.